elasticsearch/src/main/java/org/hibernate/search/elasticsearch/cfg/IndexSchemaManagementStrategy.java

```diff
@@ -49,11 +53,42 @@
 	 * <p>Note that whenever a search factory is altered after initialization (i.e. new entities are mapped),
 	 * the index will <strong>not</strong> be deleted again: new mappings will simply be added to the index.
 	 */
-	RECREATE,
+	DROP_AND_CREATE("drop-and-create"),
 
 	/**
-	 * The same as {@link #RECREATE}, but indexes - and all their contents - will be deleted upon session factory
+	 * The same as {@link #DROP_AND_CREATE}, but indexes - and all their contents - will be deleted upon session factory
 	 * shut-down as well.
 	 */
-	RECREATE_DELETE;
+	DROP_AND_CREATE_AND_DROP("drop-and-create-and-drop");
+
+	private static final Map<String, IndexSchemaManagementStrategy> VALUES_BY_EXTERNAL_NAME;
+	static {
+		Map<String, IndexSchemaManagementStrategy> tmpMap = new HashMap<>();
+		for ( IndexSchemaManagementStrategy strategy : values() ) {
+			tmpMap.put( strategy.externalName, strategy );
+		}
+		VALUES_BY_EXTERNAL_NAME = Collections.unmodifiableMap( tmpMap );
+	}
+
+	public static IndexSchemaManagementStrategy interpretPropertyValue(String propertyValue) {
+		IndexSchemaManagementStrategy strategy = VALUES_BY_EXTERNAL_NAME.get( propertyValue );
```

Comment:
I can just trim and lowercase the property value before passing it to `.get()`, it will achieve the same effect. Will do.